packages/types/src/stellar/interfaces.ts

```diff
@@ -1,67 +1,64 @@
 // Copyright 2020-2024 SubQuery Pte Ltd authors & contributors
 // SPDX-License-Identifier: GPL-3.0
 
-import {SorobanRpc, Contract, xdr} from 'stellar-sdk';
-import {HorizonApi, ServerApi} from 'stellar-sdk/lib/horizon';
-import {BaseEffectRecord} from 'stellar-sdk/lib/horizon/types/effects';
+import {Contract, xdr, Horizon} from 'stellar-sdk';
 import {BlockWrapper} from '../interfaces';
 
-export type StellarBlock = Omit<ServerApi.LedgerRecord, 'effects' | 'operations' | 'self' | 'transactions'> & {
+export type StellarBlock = Omit<Horizon.ServerApi.LedgerRecord, 'effects' | 'operations' | 'self' | 'transactions'> & {
   effects: StellarEffect[];
   operations: StellarOperation[];
   transactions: StellarTransaction[];
   events: SorobanEvent[];
 };
 
 export type StellarTransaction = Omit<
-  ServerApi.TransactionRecord,
+  Horizon.ServerApi.TransactionRecord,
   'effects' | 'ledger' | 'operations' | 'precedes' | 'self' | 'succeeds'
 > & {
   effects: StellarEffect[];
-  ledger: StellarBlock;
+  ledger: StellarBlock | null;
   operations: StellarOperation[];
   events: SorobanEvent[];
 };
 
-export type StellarOperation<T extends HorizonApi.BaseOperationResponse = ServerApi.OperationRecord> = Omit<
+export type StellarOperation<T extends Horizon.HorizonApi.BaseOperationResponse = Horizon.ServerApi.OperationRecord> =
+  Omit<T, 'self' | 'succeeds' | 'precedes' | 'effects' | 'transaction'> & {
+    effects: StellarEffect[];
+    transaction: StellarTransaction | null;
+    ledger: StellarBlock | null;
+    events: SorobanEvent[];
+  };
+
+export type StellarEffect<T extends Horizon.ServerApi.EffectRecord = Horizon.ServerApi.EffectRecord> = Omit<
   T,
-  'self' | 'succeeds' | 'precedes' | 'effects' | 'transaction'
+  'operation'
 > & {
-  effects: StellarEffect[];
-  transaction: StellarTransaction;
-  ledger: StellarBlock;
-  events: SorobanEvent[];
-};
-
-export type StellarEffect<T extends BaseEffectRecord = ServerApi.EffectRecord> = Omit<T, 'operation'> & {
-  operation: StellarOperation;
-  transaction: StellarTransaction;
-  ledger: StellarBlock;
+  operation: StellarOperation | null;
+  transaction: StellarTransaction | null;
+  ledger: StellarBlock | null;
 };
 // COPIED FROM SOROBAN, due to no longer export
 export interface SorobanRpcEventResponse extends SorobanRpcBaseEventResponse {
   contractId?: Contract;
   topic: xdr.ScVal[];
   value: xdr.ScVal;
 }
+export type EventType = 'contract' | 'system' | 'diagnostic';
 
 interface SorobanRpcBaseEventResponse {
   id: string;
-  type: SorobanRpc.Api.EventType;
+  type: EventType;
   ledger: number;
   ledgerClosedAt: string;
   pagingToken: string;
   inSuccessfulContractCall: boolean;
+  txHash: string;
 }
 
 export type SorobanEvent = Omit<SorobanRpcEventResponse, 'ledger'> & {
-  value: {
-    xdr: string;
-    decoded?: string;
-  };
-  ledger: StellarBlock;
-  transaction: StellarTransaction;
-  operation: StellarOperation;
+  ledger: StellarBlock | null;
```

Comment:
Why is value removed?

Reply:
value is also provided from `SorobanRpcEventResponse`, we should using native type `xdr.ScVal`

and also use these method to encode/decode from event, 

```
export function nativeToScVal(val: any, opts?: { type: any }): xdr.ScVal;
export function scValToNative(scv: xdr.ScVal): any;
```
It didn't break anything, see test in `block.stellar.spec.ts`